java/dagger/internal/codegen/ComponentProcessor.java

```diff
@@ -93,6 +93,10 @@ public SourceVersion getSupportedSourceVersion() {
     options.addAll(CompilerOptions.SUPPORTED_OPTIONS);
     options.addAll(spiPlugins.allSupportedOptions());
     options.addAll(validationPlugins.allSupportedOptions());
+    if (Ascii.equalsIgnoreCase(
+        "enabled", System.getProperties().getProperty("dagger.gradle.incremental"))) {
```

Comment:
As mentioned on the other thread, I think this could/should be an option, not a system property. Since `init` has already been called at this point, you already know all the options that were passed by the client.

Reply:
SGTM